packages/@aws-cdk/aws-iam/lib/policy.ts

```diff
@@ -124,7 +124,18 @@ export class Policy extends Resource implements IPolicy {
         Lazy.stringValue({ produce: () => generatePolicyName(scope, resource.logicalId) }),
     });
 
-    const resource = new CfnPolicy(this, 'Resource', {
+    // this function returns `true` if the CFN resource should be included in
+    // the cloudformation template unless `force` is `true`, if the policy
+    // document is empty, the resource will not be included.
+    const self = this;
+
+    class CfnPolicyConditional extends CfnPolicy {
+      protected shouldSynthesize() {
+        return self.force || self.referenceTaken || (!self.document.isEmpty && self.isAttached);
```

Comment:
Can you explain where this came from?

Reply:
https://github.com/aws/aws-cdk/pull/9428/files#diff-e60351c0bfb7341b1a29307a3d444c5eL225